src/gui/spatialIndicator.js

```diff
@@ -0,0 +1,419 @@
+            float fx = brightness / abs(d - 1.);
+            fx = pow(fx, .5);
+            return fx;
+        }
+        
+        void main(void) {
+            vec2 uv = vUv;
+        
+            vec3 col = vec3(0.);
+            float alpha = 0.;
+            
+            // draw the ascending lines
+            for (int i = 0; i < amount; i++) {
+                float x = lines[i].x, y = lines[i].y, width = lines[i].width, height = lines[i].height;
+                float d = GlowingLine(uv, vec2(x, y), width, height);
+                col += avatarColor[0].color * d;
+                alpha += d;
+            }
+        
+            col *= .1;
+            alpha *= .1;
+            if (alpha < .35) alpha = 0.;
+        
+            // draw the fluctuating upper alpha fade out boundary
+            // float boundary = S(0.0, 0.8, 1. - uv.y);
+            // float boundary = S(0.8, 0.0, 1. - uv.y);
+            // col = mix(col, red, boundary);
+        
+            // float boundary = S(0.2, 0.0, 1. - uv.y);
+            // alpha = mix(alpha, 0., boundary);
+        
+            gl_FragColor = vec4(col, alpha);
+        }
+    `;
+
+    const innerFragmentShader = `
+    varying vec2 vUv;
+    
+    // set up color uniforms
+    struct AvatarColor {
+        vec3 color;
+        vec3 colorLighter;
+    };
+    uniform AvatarColor avatarColor[1];
+    
+    void main(void) {
+        vec3 color = avatarColor[0].color;
+        gl_FragColor = vec4(color, 0.5);
+    }
+    `;
+
+    const amount = 5;
+    let lines = [];
+
+    let color = 'rgb(0, 255, 255)', colorLighter = 'rgb(255, 255, 255)';
+    let finalColor = [{
+        color: new THREE.Color(color),
+        colorLighter: new THREE.Color(colorLighter)
+    }];
+    let uniforms = {
+        'avatarColor': {value: finalColor},
+        'amount': {value: amount},
+        'lines': {value: lines},
+    };
+
+    const cylinderMaterial = new THREE.ShaderMaterial({
+        vertexShader: vertexShader,
+        fragmentShader: cylinderFragmentShader,
+        uniforms: uniforms,
+        transparent: true,
+        side: THREE.DoubleSide,
+    });
+    
+    const innerMaterial = new THREE.ShaderMaterial({
+        vertexShader: vertexShader,
+        fragmentShader: innerFragmentShader,
+        uniforms: uniforms,
+        transparent: true,
+        blending: THREE.CustomBlending,
+        blendEquation: THREE.AddEquation,
+        blendSrc: THREE.SrcColorFactor,
+        blendDst: THREE.OneMinusSrcAlphaFactor,
+        side: THREE.DoubleSide,
+    })
+
+    const Clamp = (x, low, high) => {
+        return Math.min(Math.max(x, low), high);
+    }
+
+    const Remap01 = (x, low, high) => {
+        return Clamp((x - low) / (high - low), 0, 1);
+    }
+
+    const Remap = (x, low1, high1, low2, high2) => {
+        return low2 + (high2 - low2) * Remap01(x, low1, high1);
+    }
+
+    window.addEventListener('pointerdown', (e) => {
+        if (e.button === 0) handleMouseClick(e);
```

Comment:
You might be able to use `!realityEditor.device.isMouseEventCameraControl(e)` instead of `e.button === 0`, to make more consistent with other clicking logic, unless you have a specific reason for checking this way instead